modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/ContentCompressionHandler.java

```diff
@@ -0,0 +1,142 @@
+     */
+    public static Boolean isCompressionEnabled = AndesConfigurationManager.readValue
+            (AndesConfiguration.PERFORMANCE_TUNING_ALLOW_COMPRESSION);
+
+    /**
+     * Maximum content size to check before compress message content, in bytes. Messages with
+     * content less than this size will not compress.
+     */
+    public static int contentCompressionThreshold = AndesConfigurationManager.readValue
+            (AndesConfiguration.PERFORMANCE_TUNING_CONTENT_COMPRESSION_THRESHOLD);
+
+    /**
+     * Getting lz4 instance
+     */
+    static LZ4Factory factory = LZ4Factory.fastestInstance();
+
+    /**
+     * Getting lz4 fast compressor to compress data
+     */
+    static LZ4Compressor compressor = factory.fastCompressor();
+
+    /**
+     * Getting lz4 fast decompressor to decompress data
+     */
+    static LZ4FastDecompressor decompressor = factory.fastDecompressor();
+
+    private static Log log = LogFactory.getLog(ContentCompressionHandler.class);
+
+    /**
+     * Compress the content of the message
+     *
+     * @param partList              Original content chunk list
+     * @param originalContentLength Total content length of the original message content
+     * @return Compressed message content as a byte array
+     */
+    public static byte[] getCompressedMessage(List<AndesMessagePart> partList, int originalContentLength) {
+
+        byte[] messageData = getByteArrayFromPartList(partList, originalContentLength);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Data before compression : " + (new String(messageData, StandardCharsets.UTF_8)));
+        }
+
+        // Compress message content
+        int maxCompressedLength = compressor.maxCompressedLength(originalContentLength);
+        byte[] compressed = new byte[maxCompressedLength];
+        int compressedLength = compressor.compress(messageData, 0, originalContentLength, compressed, 0, maxCompressedLength);
+        byte[] finalCompressedArray = Arrays.copyOf(compressed, compressedLength);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Compressed data: " + (new String(finalCompressedArray, StandardCharsets.UTF_8)));
+        }
+
+        return finalCompressedArray;
+    }
+
+    /**
+     * Decompress the compressed message content
+     *
+     * @param partList              Content chunk list of the decompressed message content
+     * @param originalContentLength Total content length of the original message content
+     * @return Decompressed message content as a byte array
+     */
+    public static byte[] getDecompressedMessage(List<AndesMessagePart> partList, int originalContentLength) {
+        byte[] compressedMessageContent = getByteArrayFromPartList(partList, originalContentLength);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Data before decompression : " + (new String(compressedMessageContent, StandardCharsets.UTF_8)));
```

Comment:
any reason why we do this? ( for my knowledge). and why do we need to print the content.?